dkist/dataset/tiled_dataset.py

```diff
@@ -82,8 +82,8 @@ def _from_components(cls, shape, file_managers, wcses, header_tables, inventory)
 
         return cls(datasets, inventory)
 
-    def __init__(self, dataset_array, inventory=None):
-        self._data = np.array(dataset_array, dtype=object)
+    def __init__(self, dataset_array, inventory=None, mask=False):
+        self._data = np.ma.masked_array(dataset_array, dtype=object, mask=mask)
```

Comment:
Can we auto-generate the mask based on the elements of dataset_array?

Reply:
I expect so. We can readily assume anything that isn't a Dataset should be masked out, but we should probably also have some way of flagging invalid Datasets for masking as well.

Reply:
Shouldn't the default value of mask be `None`?

Reply:
Either works fine. `False` is specifying  to set the mask to be `False` (ie: not masked) everywhere. Which is the default if you pass `None` into `masked_array` anyway, but explicit is better than implicit.

Reply:
If we set it to `None` that could be "auto" where as `False` would be no-mask.